springboot/src/main/kotlin/com/goliath/emojihub/springboot/dao/UserDao.kt

```diff
@@ -15,14 +17,37 @@ class UserDao {
         const val COLLECTION_NAME = "Users"
     }
 
-    fun getUsers(): List<User> {
-        val list = mutableListOf<User>()
+    fun getUsers(): List<UserDto> {
+        val list = mutableListOf<UserDto>()
         val db: Firestore = FirestoreClient.getFirestore()
         val future = db.collection(COLLECTION_NAME).get()
         val documents: List<QueryDocumentSnapshot> = future.get().documents
         for (document in documents) {
-            list.add(document.toObject(User::class.java))
+            list.add(document.toObject(UserDto::class.java))
         }
         return list
     }
+
+    fun getUser(username: String): UserDto? {
+        var userDto: UserDto? = null
+        val db: Firestore = FirestoreClient.getFirestore()
+        val future = db.collection(COLLECTION_NAME).document(username).get()
+        val document: DocumentSnapshot = future.get()
+        if (document.exists()) {
+            userDto = document.toObject(UserDto::class.java)!!
+        }
+        return userDto
+    }
```

Comment:
`!!` 연산자 사용의 지양 + Early return pattern + `UserServiceImpl`에서 null에 대한 처리가 이루어짐을 감안하여 다음과 같이 작성하는건 어떨지 제안 드려봅니다 😀 굳이? 싶다면 넘겨도 괜찮습니다
```suggestion
    fun getUser(username: String): UserDto? {
        val db: Firestore = FirestoreClient.getFirestore()
        val future = db.collection(COLLECTION_NAME).document(username).get()
        val document: DocumentSnapshot = future.get()
        if (document.exists()) {
            return document.toObject(UserDto::class.java)
        }
        return null
    }
```

Reply:
반영해 푸시하겠습니다 :+1: